napi_from_js.md

```diff
@@ -0,0 +1,204 @@
+- `napi_ok` if the API succeeded.
+
+### *napi_get_cb_args*
+
+#### Signature
+```
+napi_status napi_get_cb_args(napi_env env,
+                             napi_callback_info cbinfo,
+                             napi_value* buffer,
+                             size_t bufferlength)
+```
+
+#### Parameters
+- `[in] e`: The environment that the API is invoked under
+- `[in] cbinfo`: The callback info passed into the callback function
+- `[in] buffer`: Array of `napi_value` into which the arguments to the 
+callback function will be copied. This array must be user-allocated
+and have space to store at least `bufferlength`-count elements
+- `[in] bufferlength`: The size in elements in the buffer array
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_cb_data*
+
+#### Signature
+```
+napi_status napi_get_cb_data(napi_env env,
+                             napi_callback_info cbinfo,
+                             void** result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[out] result`: The pointer to the `data` parameter that was passed
+in when the callback was initially installed
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_cb_this*
+
+#### Signature
+```
+napi_status napi_get_cb_this(napi_env env,
+                             napi_callback_info cbinfo,
+                             napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[out] result`: The `napi_value` representing the value of `this` 
+provided for the call to the function.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_cb_args_length*
+
+#### Signature
+```
+napi_status napi_get_cb_args_length(napi_env env,
+                                    napi_callback_info cbinfo,
+                                    size_t* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[out] result`: The number of actual arguments passed into the function
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_cb_info*
+
+#### Signature
+```
+napi_status napi_get_cb_info(napi_env env,
+                             napi_callback_info cbinfo,
+                             size_t* argc, 
+                             napi_value* argv,
+                             napi_value* thisArg, 
+                             void** data)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[in-out] argc`: Specifies the size of the provided argv array
+and receives the actual count of args.
+- `[out] argv`: Buffer to which the `napi_value` representing the
```

Comment:
Notes: If there are actually more arguments than the provided argument count, then only the requested number are copied. If there are fewer, then the argv is filled in with `undefined`.